src/storage/block_metadata.h

```diff
@@ -0,0 +1,48 @@
+//
+//
+//
+
+#ifndef HUSTLE_BLOCK_METADATA_H
+#define HUSTLE_BLOCK_METADATA_H
+
+namespace hustle::storage {
+
+class BlockMetadata {
+ public:
+  /**
```

Comment:
can this 2 (isokay and getstatus) be combined into 1 function? we can have getstatus return a construction error